netty/src/test/java/io/grpc/netty/ProtocolNegotiatorsTest.java

```diff
@@ -415,7 +415,7 @@ public void from_tls_clientAuthRequire_noClientCert() throws Exception {
         .trustManager(caCert)
         .build();
     Status status = expectFailedHandshake(channelCreds, serverCreds);
-    assertThat(status.getDescription()).isEqualTo("ssl exception");
+    assertThat(status.getDescription()).containsMatch("(^ssl exception|SslHandler)");
```

Comment:
In what case it is "ssl exception" and in what other case it is "SslHandler"? The unit test should be as deterministic as possible. Seems we can't simply leave the unit test like this.

Reply:
`SslException` is how it was and is still the case for Linux. However for MacOS and Windows it generates an IOException with the description containing `"SslHandler"` . This difference in behavior is probably due to the new netty-tcnative library. Not sure how else to address this difference in behavior.

Reply:
@ejona86 Is this what the test expects?

If the new netty-tcnative library behaves differently in different OS's, should we have different assertions based on OS name Sys property?

Reply:
@ejona86 @sergiitk @dapengzhang0 the logic is now changed as per our latest discussion and copied from https://github.com/grpc/grpc-java/pull/7766/files#diff-db452b0dac29b1f1e09c876f2cf70c9529fb46caa19bdd55c6d6a454401f9624R329

Reply:
Hopefully this PR is now ready to be merged